bin/reth/src/commands/debug_cmd/replay_engine.rs

```diff
@@ -196,7 +197,14 @@ impl Command {
         let (consensus_engine_tx, consensus_engine_rx) = mpsc::unbounded_channel();
         let (beacon_consensus_engine, beacon_engine_handle) = BeaconConsensusEngine::with_channel(
             network_client,
-            Pipeline::builder().build(provider_factory),
+            Pipeline::builder().build(
+                provider_factory.clone(),
+                StaticFileProducer::new(
+                    provider_factory.clone(),
+                    provider_factory.static_file_provider(),
+                    PruneModes::default(),
+                ),
+            ),
```

Comment:
would like to improve the builder pattern here such that we don't have 3 mentions of provider factory in the call

Reply:
it's a bit of an unreal example because of debug cmd, in practice, it looks like this
https://github.com/paradigmxyz/reth/blob/0a6e95c21b4d131f41d5bb9c7f33a5a15c22f3dd/crates/node-builder/src/builder.rs#L359-L364
https://github.com/paradigmxyz/reth/blob/a1df62c1ad7721f67857f100b28e3ecfeb7c2e8d/crates/node-core/src/node_config.rs#L890

but I agree that we can pass only `provider_factory` and `prune_modes` inside the `StaticFileProducer::new`